core/src/main/java/org/elasticsearch/search/aggregations/metrics/avg/AvgAggregator.java

```diff
@@ -76,15 +79,22 @@ public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
             public void collect(int doc, long bucket) throws IOException {
                 counts = bigArrays.grow(counts, bucket + 1);
                 sums = bigArrays.grow(sums, bucket + 1);
+                compensations = bigArrays.grow(compensations, bucket + 1);
 
                 if (values.advanceExact(doc)) {
                     final int valueCount = values.docValueCount();
                     counts.increment(bucket, valueCount);
-                    double sum = 0;
+                    double sum = sums.get(bucket);
+                    double compensation = compensations.get(bucket);
+
                     for (int i = 0; i < valueCount; i++) {
-                        sum += values.nextValue();
+                        double corrected = values.nextValue() - compensation;
+                        double newSum = sum + corrected;
+                        compensation = (newSum - sum) - corrected;
+                        sum = newSum;
```

Comment:
can you add a comment saying this is Kahan summation?

Reply:
Thanks @jpountz . I pushed 6c45e0874490e87310680b8f699f97a50beb6045